datadog/resource_datadog_integration_test.go

```diff
@@ -0,0 +1,119 @@
+package datadog
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	datadog "github.com/zorkian/go-datadog-api"
+)
+
+// We're not testing for schedules because Datadog actively verifies it with Pagerduty
+
+func TestAccDatadogIntegrationPagerduty_Basic(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckDatadogIntegrationPagerdutyDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckDatadogIntegrationPagerdutyConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDatadogIntegrationPagerdutyExists("datadog_integration_pagerduty.foo"),
+					resource.TestCheckResourceAttr(
+						"datadog_integration_pagerduty.foo", "subdomain", "testdomain"),
+					resource.TestCheckResourceAttr(
+						"datadog_integration_pagerduty.foo", "api_token", "*****"),
+					resource.TestCheckResourceAttr(
+						"datadog_integration_pagerduty.foo", "services.0.service_name", "test_service"),
+					resource.TestCheckResourceAttr(
+						"datadog_integration_pagerduty.foo", "services.0.service_key", "*****"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccDatadogIntegrationPagerduty_TwoServices(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckDatadogIntegrationPagerdutyDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckDatadogIntegrationPagerdutyConfig_TwoServices,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDatadogIntegrationPagerdutyExists("datadog_integration_pagerduty.foo"),
+					resource.TestCheckResourceAttr(
+						"datadog_integration_pagerduty.foo", "subdomain", "testdomain"),
+					resource.TestCheckResourceAttr(
+						"datadog_integration_pagerduty.foo", "api_token", "*****"),
+					resource.TestCheckResourceAttr(
```

Comment:
Does the API guarantee stable field ordering? If not this test will be flaky

Reply:
The API doc doesn't guarantee stable field ordering but from what I've seen in my usage/tests. They've been consistent.